### ISSUE TITLE
Rename transects siteID and transectID

### ISSUE THREAD

**Issue by jarad** [jarad/ISUmonarch#61]
The pre transects currently break the naming convention for transectIDs. The transectID naming convention is "t" followed by 4 digit siteID (3 letters followed by a number) followed by a letter indicating which transect it is at that site. This last letter increments with each new transect. 

Currently the pre transects have the same format, but the last letter indicates a different transect rather than a different location. Conversely other transects, e.g. byd, the number indicates a different transect at the same location. 

Ideally, we have a rule for creating siteID and transectID and follow that rule.

**Comment by sethapp** [comment]
I believe these are the rules I want us to follow

- The three letter prefix indicates the site. Plots at the same site are 1) within 1 mile of each other and 2) owned by the same entity but 3) plot variables may differ greatly (preexisting vegetation, slope, elevation, etc)
- There is always one transect associated with one plot but there may be more than one plot/transect combination at a site (as indicated by them sharing the same three letter prefix)
- The number following the three letter prefix indicates the plot number at the site
- The letter following the plot number indicates whether the the plot/transect combination has ever changed location. If the plot/transect combination has never been moved the letter is ‘a’. Every time a plot/transect combination moves the transect_id final letter is changed to the next letter in the alphabet.
- a ‘t’ preceding the transect name indicates that this is a transect_id

 The application of these would produce the following renaming

- > _van2 | tvan2a | Van Diest 2_ (old)

is now `dav1 | tdav1a | Davison (replaces van2)`

- These will be renamed like this:
pre1 | tpre1a | Prestage 1
pre2 | tpre2a | Prestage 2
pre3 | tpre3a | Prestage 3
pre4 | tpre4a | Prestage 4
pre5 | tpre5a | Prestage 5
pre6 | tpre6a | Prestage 6
pre7 | tpre7a | Prestage 7
pre8 | tpre8a | Prestage 8

**Comment by jarad** [comment]
Sounds good.

**Comment by nulloa** [comment]
OK so I've updated @sethapp 's comment to make a little more sense to myself. This will be done using [http://www.jarad.me/resources/utility.html](http://www.jarad.me/resources/utility.html) and possibly using `R` directly.

**Comment by nulloa** [comment]
A quick question, we wanted to change everything with this structure correct? csv, csv2, pdf? @sethapp @jarad

**Comment by sethapp** [comment]
Yes, I believe so

**Comment by nulloa** [comment]
Ok, one last quick question, @sethapp you know that those names are already in use right?
```
> unique(nectar[grep("pre", nectar$siteID),c("siteID", "transectID")])
     siteID transectID
790   pre1a     tpre1a
798   pre1b     tpre1b
808   pre2b     tpre2b
813   pre4a     tpre4a
834   pre2a     tpre2a
844   pre3b     tpre3b
1089  pre3a     tpre3a
1467  pre4b     tpre4b
3372   pre2     tpre2b
3377   pre3     tpre3a
3382   pre4     tpre4a
3386   pre4     tpre4b
3394   pre1     tpre1a
3398   pre1     tpre1b
3403   pre2     tpre2a
3416   pre3     tpre3b
```

Because we'll be combining into obs that already have the same `siteID` and `transectID`

**Comment by sethapp** [comment]
I understand that all site names pre1a-pre4a (only the a's) and transect names tpre1a-tpre4a (only the a's) are already in use. 

I don't know why there are sites that didn't contain the a or b. 
![image](https://user-images.githubusercontent.com/22353630/53757910-7e786200-3e82-11e9-8162-f87f405b4b64.png)
this would have been a mistake I believe. 

I understand that some site/transect names will be 'overwritten'

**Comment by nulloa** [comment]
So here is a snippet of the data using those:
```
> nectar %>% filter(siteID %in% paste0("pre", 1:4))
# A tibble: 179 x 10
    year month day   siteID transectID round section observer      `Nectar Plant Species` count
   <dbl> <chr> <chr> <fct>  <fct>      <chr> <chr>   <chr>         <chr>                  <int>
 1  2016 07    01    pre2   tpre2b     1     0-10m   SteveBKelseyF common dandelion           0
 2  2016 07    01    pre2   tpre2b     1     10-20m  SteveBKelseyF common dandelion           0
 3  2016 07    01    pre2   tpre2b     1     20-30m  SteveBKelseyF common dandelion           0
 4  2016 07    01    pre2   tpre2b     1     30-40m  SteveBKelseyF common dandelion           1
 5  2016 07    01    pre2   tpre2b     1     40-50m  SteveBKelseyF common dandelion           0
 6  2016 07    01    pre3   tpre3a     1     0-20m   SteveBCoryH   common dandelion           0
 7  2016 07    01    pre3   tpre3a     1     20-40m  SteveBCoryH   common dandelion           1
 8  2016 07    01    pre3   tpre3a     1     40-60m  SteveBCoryH   common dandelion           1
 9  2016 07    01    pre3   tpre3a     1     60-80m  SteveBCoryH   common dandelion           0
10  2016 07    01    pre3   tpre3a     1     80-100m SteveBCoryH   common dandelion           1
# … with 169 more rows
```

**Comment by sethapp** [comment]
Sorry, I could have tried to go find that myself. It appears that the site ID for 2016 (for all? or maybe most) did not include the a or b in the name. It appears that was added in 2017. The pdf data sheets don't appear to list the a or b in the site name.

**Comment by jarad** [comment]
Update the filenames so that the siteID includes the letter at the end and is consistent with the transect for those siteIDs listed above. The resulting filenames should have these new siteIDs.

    nectar$siteID <- gsub("t","",nectar$transectID)

**Comment by nulloa** [comment]
Closed with f3a89b3bf973f8eace9f8f94814f089082a4e0d2
